packages/react-reconciler/src/ReactFiber.js

```diff
@@ -779,8 +787,15 @@ export function createFiberFromFragment(
   mode: TypeOfMode,
   lanes: Lanes,
   key: null | string,
+  hasRef: boolean,
 ): Fiber {
   const fiber = createFiber(Fragment, elements, key, mode);
+  if (enableFragmentRefs && hasRef) {
+    const instance: FragmentState = {
+      ref: null,
+    };
+    fiber.stateNode = instance;
```

Comment:
It can also get a ref conditionally added later `<Fragment ref={c ? ref : undefined}>`. You could just create this in the commit phase right before sending the ref.

See how ViewTransitions does this.

https://github.com/facebook/react/blob/main/packages/react-reconciler/src/ReactFiberCommitEffects.js#L885-L887

Now in this case you don't need a wrapper state object so you can just use `stateNode` for this.